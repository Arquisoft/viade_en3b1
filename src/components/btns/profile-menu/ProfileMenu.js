import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import { MenuItem, Menu, Link, Avatar } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { withStyles } from '@material-ui/core/styles';
import { Value } from '@solid/react';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import PersonIcon from '@material-ui/icons/Person';
import PeopleIcon from '@material-ui/icons/People';
import { GetUserProfileImage } from '../../../handler/UserDataHandler';
import ContactSupportIcon from '@material-ui/icons/ContactSupport';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import { LogOut } from '../../../handler/SessionHandler';
import SupervisedUserCircleIcon from '@material-ui/icons/SupervisedUserCircle';
import InfoIcon from '@material-ui/icons/Info';

const ProfileMenu = () => {

    const [url, setUrl] = useState("");

    GetUserProfileImage().then((path) => {
        setUrl(path);
    });

    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState(null);

    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
    };

    const handleClose = () => {
        setAnchorEl(null);
    };

    return (
        <div>
            <Button
                aria-controls="customized-menu"
                aria-haspopup="true"
                variant="contained"
                color="primary"
                onClick={handleClick}
                className={classes.button}
            >

                <Avatar alt="Profile photo" src={url} className={classes.profile} /> <ArrowDropDownIcon />
            </Button>
            <StyledMenu
                id="customized-menu"
                anchorEl={anchorEl}
                keepMounted
                open={Boolean(anchorEl)}
                onClose={handleClose}
            >
                <StyledMenuItem divider disableRipple >
                    <ListItemIcon>
                        <Avatar alt="Profile photo" src={url} className={classes.profileBig} />
                    </ListItemIcon>
                    <ListItemText primary={<Value src="user.name" />} />
                </StyledMenuItem>

                <Link underline='none' href={'#/profile'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <PersonIcon />
                        </ListItemIcon>
                        <ListItemText primary="Your profile" />
                    </StyledMenuItem>
                </Link>

                <Link underline='none' href={'#/profile/friends'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <PeopleIcon />
                        </ListItemIcon>
                        <ListItemText primary="Friends" />
                    </StyledMenuItem>
                </Link>

                <Link underline='none' href={'#/profile/groups'}>
                    <StyledMenuItem divider>
                        <ListItemIcon>
                            <SupervisedUserCircleIcon />
                        </ListItemIcon>
                        <ListItemText primary="Groups" />
                    </StyledMenuItem>
                </Link>

                <Link underline='none' href={'#/help'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <ContactSupportIcon />
                        </ListItemIcon>
                        <ListItemText primary="Need help?" />
                    </StyledMenuItem>
                </Link>

                <Link underline='none' href={'#/about'}>
                    <StyledMenuItem divider>
                        <ListItemIcon>
                            <InfoIcon />
                        </ListItemIcon>
                        <ListItemText primary="About us" />
                    </StyledMenuItem>
                </Link>

                <Link underline='none' onClick={() => LogOut()}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <ExitToAppIcon />
                        </ListItemIcon>
                        <ListItemText primary="Sign out" />
                    </StyledMenuItem>
                </Link>
            </StyledMenu>
        </div>
    );
};

const StyledMenu = withStyles({
    paper: {
        border: '1px solid #d3d4d5',
    },
})((props) => (
    <Menu
        elevation={0}
        getContentAnchorEl={null}
        anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'center',
        }}
        transformOrigin={{
            vertical: 'top',
            horizontal: 'center',
        }}
        {...props}
    />
));

const StyledMenuItem = withStyles((theme) => ({
    root: {
        '&:focus': {
            backgroundColor: theme.palette.secondary.main,
        },
    },
}))(MenuItem);

const useStyles = makeStyles((theme) => ({
    button: {
        marginRight: theme.spacing(2),
    },
    profile: {
        width: theme.spacing(3),
        height: theme.spacing(3),
    },
    profileBig: {
        width: theme.spacing(5),
        height: theme.spacing(5),
    }
}));

export default ProfileMenu;