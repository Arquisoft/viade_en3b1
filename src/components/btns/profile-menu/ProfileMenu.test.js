import ProfileMenu from './ProfileMenu';

test('Define ProfileMenu', () => {
  expect(ProfileMenu).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<ProfileMenu />);
  expect(wrapper).toMatchSnapshot();
});