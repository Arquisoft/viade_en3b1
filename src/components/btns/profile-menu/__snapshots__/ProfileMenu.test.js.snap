// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <button
    aria-controls="customized-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-profile-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
       
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </span>
  </button>
</div>
`;
